src/Parquet.chpl

```diff
@@ -180,4 +184,238 @@ module Parquet {
+                      pqLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
+                      hadError = true;
+                      if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
+                  } catch e: DatasetNotFoundError {
+                      fileErrorMsg = "Dataset %s not found in file %s".format(dsetname,fname);
+                      pqLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
+                      hadError = true;
+                      if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
+                  } catch e: SegArrayError {
+                      fileErrorMsg = "SegmentedArray error: %s".format(e.message());
+                      pqLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
+                      hadError = true;
+                      if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
+                  } catch e : Error {
+                      fileErrorMsg = "Other error in accessing file %s: %s".format(fname,e.message());
+                      pqLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
+                      hadError = true;
+                      if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
+                  }
+
+                  // This may need to be adjusted for this all-in-one approach
+                  if hadError {
+                    // Keep running total, but we'll only report back the first 10
+                    if fileErrorCount < 10 {
+                      fileErrors.append(fileErrorMsg.replace("\n", " ").replace("\r", " ").replace("\t", " ").strip());
+                    }
+                    fileErrorCount += 1;
+                  }
+              }
+              // This is handled in the readFilesByName() function
+              var subdoms: [filedom] domain(1);
+              var len: int;
+              var nSeg: int;
+              len = + reduce sizes;
+
+              // Only integer is implemented for now, do nothing if the Parquet
+              // file has a different type
+              if ty == ArrowTypes.int64 || ty == ArrowTypes.int32 {
+                var entryVal = new shared SymEntry(len, int);
+                readFilesByName(entryVal.a, filenames, sizes, dsetname);
+                var valName = st.nextName();
+                st.addEntry(valName, entryVal);
+                rnames.append((dsetname, "pdarray", valName));
+              }
+          }
+
+          repMsg = _buildReadAllMsgJson(rnames, false, 0, fileErrors, st);
+          pqLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),repMsg);
+          return new MsgTuple(repMsg,MsgType.NORMAL);
+      }
+  }
+
+  proc toparquetMsg(cmd: string, payload: string, st: borrowed SymTab): MsgTuple throws {
+    if !hasParquetSupport {
```

Comment:
At the moment I think we should leave it as is since Parquet support is looped into the CI workflow.  I've been toying with modifying the config file to be either in `yaml` or `ini` format to gives us more options in the build process.  Right now there are two configuration points for Parquet (and HDF5), part in the `Makefile` and the other part in `ServerModules.cfg`.  Eventually we'll get one to work off of the other.

Reply:
I'm not so sure this makes sense anymore since Arkouda won't compile if this module is included anyway, so I don't think we get much value here anymore. The other, non-message functions would also need to be wrapped in this `!hasParquetSupport` thing for this to work as intended I think.